File: pde/problem1/problem_ii.py
import math

from FDM_heat import *

if __name__ == "__main__":
    L = math.pi
    t_max = (1, 2, 2)
    alpha = 1
    def T0(x):
        return math.sin(x)
    a = T0(0)
    b = T0(L)
    x_mesh = (10, 50, 60)
    t_mesh = (10, 1000, 1000)

    for i in range(3):
        solve_heat_equation_plot(L, t_max[i], alpha, a, b, T0, x_mesh[i], t_mesh[i])

File: pde/problem1/Problem_ii.m
function Problem_ii()
    L = pi;
    t_max = [1, 2, 2];
    alpha = 1;
    a = 0;
    b = 0;
    x_mesh = [10, 50, 60];
    t_mesh = [10, 1000, 1000];
    for i = 1:3
        subplot(2, 2, i)
        solve_heat_equation(L, t_max(i), alpha, a, b, @sin, x_mesh(i), t_mesh(i));
        title( ...
            "t\_max = " + t_max(i) + ", " + ...
            "x\_mesh = " + x_mesh(i) + ", " + ...
            "t\_mesh = " + t_mesh(i));
    end
end